apps/settings/test/unit/modules/bluetooth/bluetooth_context_test.js

```diff
@@ -1141,6 +1194,86 @@ suite('BluetoothContext', function() {
   });
 
   suite('_findDeviceByAddress > ', function() {
+    suite('found out an existed device ', function() {
+      var inputOptions, mockPairedDvices, mockRemoteDevices;
+      setup(function() {
+        inputOptions = {
+          address: '00:11:22:AA:BB:03',
+          paired: true
+        };
+        mockPairedDvices = observableArray([
+          {address: '00:11:22:AA:BB:03',
+           name: 'device-03',
+           paired: true},
+          {address: '00:21:22:AA:BB:03',
+           name: 'device-04',
+           paired: true}]);
+
+        mockRemoteDevices = observableArray([
+          {address: 'AA:BB:CC:00:11:01',
+           name: 'device-01',
+           paired: false},
+          {address: 'AA:BB:CC:00:11:02',
+           name: 'device-02',
+           paired: false}]);
+
+        this.sinon.stub(btContext,
+          'getPairedDevices').returns(mockPairedDvices);
+        this.sinon.stub(btContext,
+          'getRemoteDevices').returns(mockRemoteDevices);
+      });
+
+      test('getPairedDevices() should be called ' +
+           'getRemoteDevices() should not be called ' +
+           'and return expected device item ', function() {
+        var foundOutDevice = btContext._findDeviceByAddress(inputOptions);
+        assert.isTrue(btContext.getPairedDevices.called);
+        assert.isFalse(btContext.getRemoteDevices.called);
+        assert.equal(foundOutDevice, mockPairedDvices.get(0));
+      });
+    });
+
+    suite('did not find out match device ', function() {
+      var inputOptions, mockPairedDvices, mockRemoteDevices;
+      setup(function() {
+        inputOptions = {
+          address: '00:11:22:AA:BB:04',
+          paired: false
+        };
+        mockPairedDvices = observableArray([
```

Comment:
The array seems the same as the ones above. We can init the array in one place, maybe in `suiteSetup` of the outer suite.